src/bentoml/_internal/bento/build_config.py

```diff
@@ -544,15 +546,25 @@ def write_to_bento(self, bento_fs: FS, build_ctx: str) -> None:
             f.write(install_sh)
 
         if self.requirements_txt is not None:
-            requirements_txt_file = resolve_user_filepath(
-                self.requirements_txt, build_ctx
+            base_txt, nested_txts = find_nested_requirements_txts(
+                Path(
+                    resolve_user_filepath(self.requirements_txt, build_ctx),
+                )
             )
             copy_file_to_fs_folder(
-                requirements_txt_file,
+                str(base_txt.absolute()),
                 bento_fs,
                 py_folder,
                 dst_filename="requirements.txt",
             )
+            for txt in nested_txts:
+                dst_path = txt.relative_to(base_txt.parent)
+                copy_file_to_fs_folder(
+                    str(txt.absolute()),
+                    bento_fs,
+                    fs.path.combine(py_folder, str(dst_path.parent)),
+                    dst_filename=dst_path.name,
+                )
```

Comment:
I propose that for requirements.txt alone, we flatten all requirements into one requirements.txt.